src/Bootstrap/Slide.elm

```diff
@@ -0,0 +1,119 @@
+module Bootstrap.Slide exposing (Config, SlideOption, SlideContent, view, config, caption, customContent, image, active, addActive, removeActive, addAttributes)
+
+{-| helper module for slides
+-}
+
+import Html exposing (div, text)
+import Html.Attributes as Attributes exposing (class)
+import Html.Keyed as Keyed
+
+
+{-| Config type
+-}
+type Config msg
+    = Config
+        { options : List (SlideOption msg)
+        , content : SlideContent msg
+        , caption : Maybe { attributes : List (Html.Attribute msg), children : List (Html.Html msg) }
+        }
+
+
+type SlideOption msg
+    = Active
+    | Attrs (List (Html.Attribute msg))
+
+
+active =
+    Active
+
+
+addActive (Config settings) =
+    Config { settings | options = settings.options ++ [ Active ] }
+
+
+removeActive (Config settings) =
+    Config { settings | options = List.filter (\option -> option /= Active) settings.options }
+
+
+addAttributes newAttributes (Config settings) =
+    Config { settings | options = settings.options ++ [ Attrs newAttributes ] }
+
+
+slideAttributes : List (SlideOption msg) -> List (Html.Attribute msg)
+slideAttributes options =
+    let
+        optionToAttribute option =
+            case option of
+                Active ->
+                    [ class "active" ]
+
+                Attrs attrs ->
+                    attrs
+    in
+        [ class "carousel-item" ]
+            ++ List.concatMap optionToAttribute options
+
+
+config : List (SlideOption msg) -> SlideContent msg -> Config msg
+config options content =
+    Config
+        { options = options
+        , content = content
+        , caption = Nothing
+        }
+
+
+caption : List (Html.Attribute msg) -> List (Html.Html msg) -> Config msg -> Config msg
+caption attributes children (Config settings) =
+    Config { settings | caption = Just { attributes = attributes, children = children } }
+
+
+
+{-
+   Slide.config [ Slide.active ]
+           (Slide.image [ Attributes.alt "slide 1" ] "https://...")
+       |> Slide.caption []
+           [ h3 [] [ text "First slide label" ]
+           , p [] []
+           ]
+       |> Slide.view
+-}
+
+
+type SlideContent msg
+    = Image { attributes : List (Html.Attribute msg), src : String }
+    | Custom { html : Html.Html msg }
+
+
+image : List (Html.Attribute msg) -> String -> SlideContent msg
+image attributes src =
+    Image { attributes = attributes, src = src }
+
+
+customContent : Html.Html msg -> SlideContent msg
+customContent html =
+    Custom { html = html }
+
+
+view : Config msg -> Html.Html msg
+view (Config { options, content, caption }) =
+    let
+        captionHtml =
+            case caption of
+                Nothing ->
+                    text ""
```

Comment:
It doesn't really matter, but I couldn't find any special casing of this in the virtual dom package from a quick scan. From I could work out it ends up here:
https://github.com/elm-lang/virtual-dom/blob/master/src/Native/VirtualDom.js#L321
and hence you would get a child node.

For my own personal interest I guess I just need to try it out, because for some things it might actually matter. Like when I json decode dom stuff from events and traverse childNodes, and extra node might make a pretty big difference if the logic is indexed bound.